cfme/tests/middleware/test_middleware_server.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+from cfme.middleware.server import MiddlewareServer
+from utils import testgen
+from utils.version import current_version
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.uncollectif(lambda: current_version() < '5.7'),
+]
+pytest_generate_tests = testgen.generate(testgen.provider_by_type, ["hawkular"], scope="function")
+
+
+def test_list_servers(provider):
+    """Tests servers lists between UI, DB and Management system
+
+    Steps:
+        * Get servers list from UI
+        * Get servers list from Database
+        * Get servers list from Management system(Hawkular)
+        * Get headers from UI
+        * Compare headers from UI with expected headers list
+        * Compare content of all the list [UI, Database, Management system]
+    """
+    ui_servers = _get_servers_set(MiddlewareServer.servers())
+    db_servers = _get_servers_set(MiddlewareServer.servers_in_db())
```

Comment:
Good question, I see the inconsistency here, but the idea of this test is to check all servers of all providers.